pluralsight/design-system
# Core: Add support for TypeScript

# Core: Add support for TypeScript

I started adding types for another `CircularProgress`, and that story depends on `Core`, so I thought it would make sense to start there.

### Design Decisions

The first commit added an `index.d.ts` following the same pattern as I did with `Tab`. The second commit takes it one step further and just converts it to TypeScript. There were no changes to the code required, just changing the extension to `.ts`, and now it uses `tsc` instead of `babel` to compile it.

If we're not comfortable taking a "production" dependency on TypeScript at this point, we could just go with the first commit. It may be tedious to keep the declaration file in sync with the actual code, but it wouldn't require any changes to the build, so that's the trade-off.

The only thing I wasn't sure about with using `tsc` was which version of EcmaScript to target. The default is `ES3`. I somewhat arbitrarily chose `ES5` but I don't think it would make a difference here.

### How to Verify

- Compare the compiled output between `babel` and `tsc`
- Ensure that running `tsc` in `npm run build` copies out `.d.ts` alongside the other `.js` files in `dist`
